### PR TITLE
Add new parameters 

### PR DESCRIPTION
**Pull request by datr93**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [ ] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json [hunk 1]
```diff
@@ -2778,6 +2806,10 @@
           "type": "boolean",
           "description": "Allow a new session."
         },
+        "virtualMachineId": {
+          "type": "string",
+          "description": "Resource Id of SessionHost's underlining virtual machine."
```
Comment (markcowl) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:2811]:
```suggestion
          "description": "Resource Id of SessionHost's underlying virtual machine."

```

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by datr93** [comment]
@markcowl sorry we need one more breaking change to review.

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json [hunk 2]
```diff
@@ -2277,6 +2170,20 @@
         "ssoContext": {
           "type": "string",
           "description": "Path to keyvault containing ssoContext secret."
+        },
+        "preferredAppGroupType": {
```
Comment (markcowl) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:2174]: If this isn't nullable, then it should be required, shouldn't it?

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-09-24-preview/desktopvirtualization.json [hunk 3]
```diff
@@ -2797,6 +2718,10 @@
           "type": "boolean",
           "description": "Allow a new session."
         },
+        "virtualMachineId": {
```
Comment (markcowl) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-09-24-preview/desktopvirtualization.json:2721]: is this a value modifiable by the user, or is it generated by the service/immutable?

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json [hunk 4]
```diff
@@ -2285,6 +2178,20 @@
         "ssoContext": {
           "type": "string",
           "description": "Path to keyvault containing ssoContext secret."
+        },
+        "preferredAppGroupType": {
```
Comment (markcowl) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:2182]: same question

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json [hunk 5]
```diff
@@ -2820,6 +2741,10 @@
           "type": "boolean",
           "description": "Allow a new session."
         },
+        "virtualMachineId": {
```
Comment (markcowl) [inline specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:2744]: see above - is this user-supplied?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json (hunk 2 above)
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-01-23-preview/desktopvirtualization.json:2174]: adding required

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-09-24-preview/desktopvirtualization.json (hunk 3 above)
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-09-24-preview/desktopvirtualization.json:2721]: its immutable adding readonly

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json (hunk 4 above)
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:2182]: adding required

File: specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json (hunk 5 above)
Comment (datr93) [reply specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2019-12-10-preview/desktopvirtualization.json:2744]: its immutable adding readonly

**Review by markcowl — APPROVED** [review]

**Review by markcowl — CHANGES_REQUESTED** [review]
Need to fix examples to include the newly required property

**Review by markcowl — APPROVED** [review]

**Pull request merged**